src/generic.rs

```diff
@@ -1470,39 +1470,43 @@ where
     R: Runtime,
 {
     runtime: PhantomData<R>,
-    tx: mpsc::Sender<PyObject>,
+    tx: mpsc::Sender<Py<PyAny>>,
 }
 
 #[cfg(feature = "unstable-streams")]
 impl<R> Sender for GenericSender<R>
 where
     R: Runtime + ContextExt,
 {
-    fn send(&mut self, py: Python, locals: TaskLocals, item: PyObject) -> PyResult<PyObject> {
+    fn send(&mut self, py: Python, locals: TaskLocals, item: Py<PyAny>) -> PyResult<Py<PyAny>> {
         match self.tx.try_send(item.clone_ref(py)) {
             Ok(_) => Ok(true.into_pyobject(py)?.into_any().unbind()),
             Err(e) => {
                 if e.is_full() {
                     let mut tx = self.tx.clone();
-                    Python::with_gil(move |py| {
+                    Python::attach(move |py| {
                         Ok(
-                            future_into_py_with_locals::<R, _, PyObject>(py, locals, async move {
-                                if tx.flush().await.is_err() {
-                                    // receiving side disconnected
-                                    return Python::with_gil(|py| {
-                                        Ok(false.into_pyobject(py)?.into_any().unbind())
-                                    });
-                                }
-                                if tx.send(item).await.is_err() {
-                                    // receiving side disconnected
-                                    return Python::with_gil(|py| {
-                                        Ok(false.into_pyobject(py)?.into_any().unbind())
-                                    });
-                                }
-                                Python::with_gil(|py| {
-                                    Ok(true.into_pyobject(py)?.into_any().unbind())
-                                })
-                            })?
+                            future_into_py_with_locals::<R, _, Py<PyAny>>(
+                                py,
+                                locals,
+                                async move {
+                                    if tx.flush().await.is_err() {
+                                        // receiving side disconnected
+                                        return Python::attach(|py| {
+                                            Ok(false.into_pyobject(py)?.into_any().unbind())
```

Comment:
While we're here, I think we can use `IntoPyObjectExt` to make this a bit cleaner, and similar below.

```suggestion
                                            false.into_py_any()
```

Reply:
[`dee585a` (#54)](https://github.com/PyO3/pyo3-async-runtimes/pull/54/commits/dee585a12c773b2380c6ff6637ef5ae1ffc052c8)